experimental/ComplexReflectionGroups/src/complex_reflection_group.jl

```diff
@@ -0,0 +1,1729 @@
+# This file implements explicit models of complex reflection groups.
+#
+# References:
+#
+# * Lehrer, G. I., & Taylor, D. E. (2009). Unitary reflection groups (Vol. 20, p. viii). Cambridge University Press, Cambridge.
+# 
+# * Thiel, U. (2014). On restricted rational Cherednik algebras. TU Kaiserslautern.
+#
+# * Marin, I., & Michel, J. (2010). Automorphisms of complex reflection groups. Represent. Theory, 14, 747–788.
+#
+# Ulrich Thiel, 2023 
+
+export complex_reflection_group
+export is_complex_reflection_group
+export complex_reflection_group_type
+export complex_reflection_group_model
+export complex_reflection_group_dual
+
+function complex_reflection_group(G::ComplexReflectionGroupType; model=:CHEVIE)
+    
+    # this will be the list of matrix groups corresponding to the components of G
+    component_groups = MatrixGroup[]
+
+    # list of models
+    modellist = []
+
+    for C in components(G)
+
+        t = C.type[1]
+        gens = nothing
+        Cmodel = nothing
+
+         # Set default model (CHEVIE)
+         if model === nothing
+            Cmodel = :CHEVIE
+        else
+            Cmodel = model
+        end
+
+        # we now create the list "gens" of matrix generators for C in the selected model
+        if isa(t, Int)
+
+            # Exceptional groups
+
+            # G4
+            if t == 4
+
+                if Cmodel == :LT
+                    # See Lehrer & Taylor (2009), page 85-86
+                    R,x = polynomial_ring(QQ)
+                    K,i = number_field(x^2+1, "i")
+                    R,x = polynomial_ring(K)
+                    K,ω = number_field(x^2+x+1, "ω")
```

Comment:
We have a general rule against unicode inside code (it is OK in comments, though), see https://docs.oscar-system.org/stable/DeveloperDocumentation/styleguide/#Unicode

Reply:
There is a user-settable flag for this. In the default case, use `omega`, but if a user opts-in to unicode, you can print `ω`. See https://docs.oscar-system.org/dev/DeveloperDocumentation/printing_details/#Unicode-printing for some documentation and e.g. https://github.com/oscar-system/Oscar.jl/pull/3270 for a real-world example of this

Reply:
And then how do I enter ω in my code when I am not supposed to use unicode in code?

Reply:
And how do I do this in this specific case for the generator of my field? (Only for this I used unicode).

Reply:
In these branches dependent on `Oscar.is_unicode_allowed()` it is fine to use unicode. See e.g. https://github.com/oscar-system/Oscar.jl/pull/3270/files#diff-11c0a2ef064e801ca05579645558af1993e8decdbbfeb2694996be0c11289fc1R926

Reply:
https://github.com/oscar-system/Oscar.jl/pull/3342/files#r1516157357

Reply:
Thanks! One question: is there anyone on this planet (or universe) who is *not* allowing unicode? 

Let's make this a rhetorical question. I fear there will be (and were) some discussions...

Reply:
> In these branches dependent on `Oscar.is_unicode_allowed()` it is fine to use unicode.

Didn't know that. I messed around with human unreadable `\u????` codes in the Young tableaux printing.

Reply:
Some people will always object, but I at least have done it myself and consciously accepted it in PRs.

@ulthiel &lt;sarcasm>no, nobody has objections to (or experiences issues due to) the use of unicode always and everywhere, we only have this settings because we were bored and wanted to spice up our lives 😈 &lt;/sarcasm> (to be clear, I've been fine using Unicode for everything and everywhere about 2001, thank you Mac OS X)

Reply:
I can't withstand: I like the is_unicode option. But then there's still unicode in the code, which will still be problematic for people using something like ed on a PDP-7. So, this is somewhat inconsistent, no?!

Anyways, if it's officially okay to put unicode in the code when wrapped around is_unicode as suggested by @lgoettgens above, I'll go with that because I like my ω and τ. 

Stuff like that seems to be more complicated than the mathematics... ;-)

Reply:
Let's not discuss further, I'm implementing as suggested by @lgoettgens.